src/client.js

```diff
@@ -292,19 +292,23 @@ Client.prototype.setPassword = function(hash, callback) {
 		client.manager.updateUser(client.name, {
 			token: token,
 			password: hash
-		});
+		}, function(err) {
+			if (err) {
+				return callback(false);
+			}
 
-		// re-read the hash off disk to ensure we use whatever is saved. this will
-		// prevent situations where the password failed to save properly and so
-		// a restart of the server would forget the change and use the old
-		// password again.
-		var user = client.manager.readUserConfig(client.name);
-		if (user.password === hash) {
-			client.config.password = hash;
-			callback(true);
-		} else {
-			callback(false);
-		}
+			// re-read the hash off disk to ensure we use whatever is saved. this will
+			// prevent situations where the password failed to save properly and so
+			// a restart of the server would forget the change and use the old
+			// password again.
+			var user = client.manager.readUserConfig(client.name);
```

Comment:
This is probably no longer necessary as the callback provides `err` from the write operation.